### PR TITLE
feat: actions to automatically build binaries

### PR DESCRIPTION
**Pull request by load1n9**

### CODE REVIEW

File: src/native/ffi.ts [hunk 1]
```diff
@@ -1,3 +1,5 @@
+import { prepare } from "https://deno.land/x/plug@0.5.2/plug.ts"
```
Comment (eliassjogreen) [inline src/native/ffi.ts:1]:
```suggestion
import { dlopen } from "https://deno.land/x/plug@1.0.0-rc.3/mod.ts"
```

[Please use the latest version of plug (1.0.0-rc.3)](https://deno.land/x/plug@1.0.0-rc.3)

File: src/native/ffi.ts [hunk 2]
```diff
@@ -165,11 +167,35 @@ const symbols = {
...
+
+const opts = {
+  name: "netsaur",
+  urls: {
+    darwin,
+    windows: uri + "libnetsaur.dll",
+    linux: uri + "libnetsaur.so",
+  },
+  policy: undefined,
+}
```
Comment (eliassjogreen) [inline src/native/ffi.ts:194]: That way this logic for different archs can be simplified

File: src/native/ffi.ts [hunk 3]
```diff
@@ -165,11 +167,35 @@ const symbols = {
...
+
+
+const mod = await prepare(opts, symbols);
```
Comment (eliassjogreen) [inline src/native/ffi.ts:197]:
```suggestion
const mod = await dlopen(opts, symbols);
```

**Review by eliassjogreen — CHANGES_REQUESTED** [review]

File: src/native/ffi.ts [hunk 4]
```diff
@@ -165,11 +167,18 @@ const symbols = {
...
 
-export default Deno.dlopen(
-  new URL(`../../native/build/${Deno.build.os === "windows" ? "" : "lib"}netsaur.${Deno.build.os === "linux" ? "so" : Deno.build.os === "darwin" ? "dylib" : "dll"}`, import.meta.url),
-  symbols,
-)
-  .symbols;
+const opts: FetchOptions = {
+  name: "netsaur",
+  url: "https://github.com/denosaurs/netsaur/releases/download/0.1.4/",
+  prefixes: {
```
Comment (eliassjogreen) [inline src/native/ffi.ts:173]: I think you want to add an `suffixes` property to properly suffix aarch64

**Review by eliassjogreen — CHANGES_REQUESTED** [review]

File: src/native/ffi.ts (hunk 4 above)
Comment (load1n9) [reply src/native/ffi.ts:173]: support for aarch64 isn't quite there yet

**Review by eliassjogreen — APPROVED** [review]

**Pull request merged**